wpe/src/main/cpp/Browser/ScopedFD.h

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (C) 2023 Igalia S.L. <info@igalia.com>
+ *   Author: Jani Hautakangas <jani@igalia.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#pragma once
+
+#include <unistd.h>
+
+class ScopedFD {
+public:
+    ScopedFD()
+        : m_fd(-1)
+    {
+    }
+    explicit ScopedFD(int fileDescriptor)
+        : m_fd(fileDescriptor)
+    {
+    }
+    ~ScopedFD()
```

Comment:
Destructor must be virtual if your class is not final.
If your class is not designed to be heritated, then better set the class final and let the destructor as this (no virtual).